pandas/tests/window/test_expanding.py

```diff
@@ -264,3 +264,25 @@ def test_expanding_skew_kurt_numerical_stability(method):
     s = s + 5000
     result = getattr(s.expanding(3), method)()
     tm.assert_series_equal(result, expected)
+
+
+@pytest.mark.parametrize("window", [1, 3, 10, 50, 1000])
+@pytest.mark.parametrize("method", ["min", "max", "average"])
+@pytest.mark.parametrize("pct", [True, False])
+@pytest.mark.parametrize("ascending", [True, False])
+@pytest.mark.parametrize("test_data", ["default", "duplicates", "nans"])
+def test_rank(window, method, pct, ascending, test_data):
+    length = 1000
+    if test_data == "default":
+        ser = Series(data=np.random.rand(length))
+    elif test_data == "duplicates":
+        ser = Series(data=np.random.choice(3, length))
+    elif test_data == "nans":
+        ser = Series(data=np.random.choice([1.0, 0.25, 0.75, np.nan], length))
```

Comment:
Can you throw an `inf` and `-inf` in here? Not really a special case I guess with how it's implemented in `rolling_rank`, but it is special cased in other rank algos and were some issues before with `inf` and `nan`, so covering here would be good